src/sources/statsd/parser.rs

```diff
@@ -133,13 +135,18 @@ fn parse_tags(input: &&str) -> Result<MetricTags, ParseError> {
     Ok(input[1..]
         .split(',')
         .map(|chunk| {
-            let pair: Vec<_> = chunk.split(':').collect();
-            let key = &pair[0];
-            // same as in telegraf plugin:
-            // if tag value is not provided, use "true"
-            // https://github.com/influxdata/telegraf/blob/master/plugins/inputs/statsd/datadog.go#L152
-            let value = pair.get(1).unwrap_or(&"true");
-            ((*key).to_owned(), (*value).to_owned())
+            // note: the behavior of StatsD if more than one colon is found (which would presumably
+            // be part of the tag value), is to remove any additional colons from the tag value.
+            // Thus Vector expects only one colon character to be present per chunk, so the find()
+            // operation locating the first position is sufficient.
+            match chunk.find(':') {
+                // the notation `tag:` is valid for statsd, the effect is an empty string value.
+                Some(colon) => (
+                    chunk[..colon].to_string(),
+                    TagValue::Value(chunk[colon + 1..].to_string()),
+                ),
+                None => (chunk.to_string(), TagValue::Bare),
+            }
```

Comment:
I _just_ learned of `str::split_once` https://doc.rust-lang.org/std/primitive.str.html#method.split_once which, like you, I have implemented by hand repeatedly. That method simplifies this to:
```suggestion
            match chunk.split_once(':') {
                // the notation `tag:` is valid for statsd, the effect is an empty string value.
                Some((prefix, suffix)) => (prefix.to_string, TagValue::Value(suffix.to_string())),
                None => (chunk.to_string(), TagValue::Bare),
            }
```

Reply:
This is awesome. Thank you for sharing that, and making the changes!